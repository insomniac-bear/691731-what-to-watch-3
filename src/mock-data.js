export const mockData = {
  films: [
    {
      id: 1,
      filmName: `Some name`,
      posterImage: `Some url`,
      previewImage: `Some url`,
      backgroundImage: `http://sume.url`,
      backgroundColor: `#000000`,
      filmSrc: `Some url`,
      filmPreview: `Some url`,
      genre: `Some genre`,
      release: 1999,
      runtime: 99,
      rating: 99,
      scoresCount: 99,
      describe: `Some description`,
      director: `Director`,
      actors: [`Actor`, `Actor`],
      isFavorite: false,
    },
    {
      id: 2,
      filmName: `Some name`,
      posterImage: `Some url`,
      previewImage: `Some url`,
      backgroundImage: `http://sume.url`,
      backgroundColor: `#000000`,
      filmSrc: `Some url`,
      filmPreview: `Some url`,
      genre: `Some genre`,
      release: 1999,
      runtime: 99,
      rating: 99,
      scoresCount: 99,
      describe: `Some description`,
      director: `Director`,
      actors: [`Actor`, `Actor`],
      isFavorite: false,
    },
    {
      id: 3,
      filmName: `Some name`,
      posterImage: `Some url`,
      previewImage: `Some url`,
      backgroundImage: `http://sume.url`,
      backgroundColor: `#000000`,
      filmSrc: `Some url`,
      filmPreview: `Some url`,
      genre: `Some genre`,
      release: 1999,
      runtime: 99,
      rating: 99,
      scoresCount: 99,
      describe: `Some description`,
      director: `Director`,
      actors: [`Actor`, `Actor`],
      isFavorite: false,
    },
    {
      id: 4,
      filmName: `Some name`,
      posterImage: `Some url`,
      previewImage: `Some url`,
      backgroundImage: `http://sume.url`,
      backgroundColor: `#000000`,
      filmSrc: `Some url`,
      filmPreview: `Some url`,
      genre: `Some genre`,
      release: 1999,
      runtime: 99,
      rating: 99,
      scoresCount: 99,
      describe: `Some description`,
      director: `Director`,
      actors: [`Actor`, `Actor`],
      isFavorite: false,
    },
  ],
  promoFilmData: {
    id: 4,
    filmName: `Some name`,
    posterImage: `Some url`,
    previewImage: `Some url`,
    backgroundImage: `http://sume.url`,
    backgroundColor: `#000000`,
    filmSrc: `Some url`,
    filmPreview: `Some url`,
    genre: `Some genre`,
    release: 1999,
    runtime: 99,
    rating: 99,
    scoresCount: 99,
    describe: `Some description`,
    director: `Director`,
    actors: [`Actor`, `Actor`],
    isFavorite: false,
  },
  comments: [
    {
      id: 0,
      user: {
        id: 2,
        name: `Author name`,
      },
      rating: 9.9,
      comment: `Some comment`,
      date: `January 01, 2001`,
    },
    {
      id: 1,
      user: {
        id: 1,
        name: `Author name`,
      },
      rating: 9.9,
      comment: `Some comment`,
      date: `January 01, 2001`,
    }

  ],
  showedFilmsCount: 8,
  filmId: 2,
  selectedGenre: `All genres`,
};
