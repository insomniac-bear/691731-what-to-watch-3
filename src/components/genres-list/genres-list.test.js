import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import NameSpace from '../../reducer/name-space.js';
import GenresList from './genres-list.jsx';
import {mockData} from '../../mock-data.js';

const mockStore = configureStore([]);

it(`Render list of Genres`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      allFilms: mockData.films,
    },
    [NameSpace.GENRE]: {
      selectedGenre: mockData.selectedGenre,
    }
  });
  const tree = renderer
    .create(
        <Provider store={store}>
          <GenresList
            onChangeGenre={jest.fn()}
          />
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
