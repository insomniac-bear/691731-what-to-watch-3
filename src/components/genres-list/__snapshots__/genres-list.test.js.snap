// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render list of Genres 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="{genre}"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="{genre}"
      onClick={[Function]}
    >
      genre-0
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="{genre}"
      onClick={[Function]}
    >
      genre-1
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="{genre}"
      onClick={[Function]}
    >
      genre-2
    </a>
  </li>
</ul>
`;
